lib/spack/spack/environment.py

```diff
@@ -450,6 +452,8 @@ def _read_manifest(self, f):
             # enable_view is False
             self._view_path = None
 
+        self.coconcretize = config_dict(self.yaml).get('concretize_together')
```

Comment:
Can you change this to use a default `self.coconcretize = config_dict(self.yaml).get('concretize_together', False)` and remove the earlier line that sets it to False in `__init__`?

Reply:
Setting the attributes in `__init__` was done on purpose. Usually I like to have all the attributes of an instance set, for the first time, directly in `__init__` (i.e. not as a side effect of another call). If you think it's better I can set it to `None` instead of `False` - which is its default value. 

For the call to `get`, setting the default as a second argument should not be necessary as that value is already set by the schema via our custom validator in:

https://github.com/spack/spack/blob/27b07811ddca7877f8a43de66b44db5b1e75e643/lib/spack/spack/schema/__init__.py#L26-L40

I'd avoid setting the `default` via `get` as the two mechanism might get out of sync. Does that make sense to you?